CMakeLists.txt

```diff
@@ -43,46 +44,27 @@ string(REGEX REPLACE "^CBMC_VERSION *= *" "" CBMC_RELEASE ${config_inc_v})
 message(STATUS "CBMC release ${CBMC_RELEASE}")
 
 find_package(Git)
-
-macro(git_revision target files_var)
+macro(git_revision target)
   if(GIT_FOUND)
-    add_custom_command(
-      OUTPUT .release_info
-      COMMAND ${CMAKE_COMMAND} -E echo_append "#define __CBMC_VERSION " > .release_info
-      COMMAND "${GIT_EXECUTABLE}" "describe" "--tags" "--always" "--long" >> .release_info
-      COMMAND ${CMAKE_COMMAND} -E echo "#define __CBMC_VERSION_str(v) \"${CBMC_RELEASE} (\" # v \")\"" >> .release_info
-      COMMAND ${CMAKE_COMMAND} -E echo "#define __CBMC_VERSION_xstr(v) __CBMC_VERSION_str(v)" >> .release_info
-      COMMAND ${CMAKE_COMMAND} -E echo "#define CBMC_VERSION __CBMC_VERSION_xstr(__CBMC_VERSION)" >> .release_info
+    add_custom_target(
```

Comment:
Nice :-) This was along the lines of what  I was thinking. This won't have the nice 'only update timestamp if the version info has actually changes' feature that you have in the Makefile version. I was wondering if we could have the same feature for cmake by:

1) create a template version.h.in file
2) Create a 'generate_release_info.cmake' script that would basically do the commands you have here, but generate the version.h file using the cmake `configure_file` command (which won't update the timestamp if the content doesn't change)
3) Have this `add_custom_target` run the `generate_release_info.cmake` script using `cmake -P` ?

I've not tried that, or fully thought it out... but maybe it will work?

Reply:
It probably does work, but CMake and I are no longer friends after yesterday's ordeal :-P

Reply:
:-) I'll see if I can work something up...